### PR TITLE
Add coverage as a dev dependency

### LINKED ISSUE #5143: Add `coverage` as a dev dependency
Tests fail if not done

### PR DESCRIPTION
**Pull request by anth-volk**
Fixes #5143.

### CODE REVIEW

**Comment by MaxGhenis** [comment]
[Full logs](https://productionresultssa0.blob.core.windows.net/actions-results/20afd972-66fb-4e8e-bc98-263c64c0f23c/workflow-job-run-78bf5d3d-47df-5be6-a66b-bcf82a771e6e/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-09-26T21%3A20%3A11Z&sig=dOR2vWBpc%2F7aEwJ%2BtBGAYB0EKQtY%2BrjCZzcmI2m9j6Y%3D&ske=2024-09-27T07%3A36%3A55Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-09-26T19%3A36%3A55Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-08-04&sp=r&spr=https&sr=b&st=2024-09-26T21%3A10%3A06Z&sv=2024-08-04) indicate it has to do with specify periods in core and variables like `marginal_tax_rate`.

@nikhilwoodruff could you take a look?

**Comment by anth-volk** [comment]
That would explain why I couldn't reproduce, I hadn't realize `core` had been updated and my env is thus out-of-date

**Comment by MaxGhenis** [comment]
Could you bump the core pin while you're here?

**Comment by anth-volk** [comment]
Sure. Any desire to keep limited below the newest for the time being?

**Comment by anth-volk** [comment]
Core pin updated

**Comment by MaxGhenis** [comment]
If we can't fix it tomorrow then yes let's hard-pin. But the update stabilizes the random take-up values, so I wouldn't want to update the app without it.

**Comment by anth-volk** [comment]
The root of this problem is a bug in `-core` from the original code itself in the test runner. I'll open a PR there and link back here.

**Comment by anth-volk** [comment]
PR for the `-core` bug opened at https://github.com/PolicyEngine/policyengine-core/pull/284.

File: setup.py [hunk 1]
```diff
@@ -34,17 +34,13 @@
         ),
     ],
     install_requires=[
-        "policyengine-core>=3.6.5",
+        "policyengine-core>=3.8.0",
```
Comment (MaxGhenis) [inline setup.py:37]:
```suggestion
        "policyengine-core>=3.8.1",
```

**Comment by MaxGhenis** [comment]
Superseded by #5053

**Pull request closed without merging**